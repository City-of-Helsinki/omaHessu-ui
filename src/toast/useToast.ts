import React from 'react';

import ToastContext, { ToastContextType } from './ToastContext';

function useToast(): ToastContextType {
  return React.useContext(ToastContext);
}

export default useToast;
