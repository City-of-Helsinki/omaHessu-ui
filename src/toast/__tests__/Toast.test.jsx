import React from 'react';
import { mount } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import { updateWrapper } from '../../common/test/testUtils';
import ToastProvider from '../ToastProvider';
import useToast from '../useToast';

function TestInvoker({ toast }) {
  const { createToast } = useToast();

  const handleClick = () => {
    createToast(toast);
  };

  return (
    <button id="create" onClick={handleClick}>
      test create
    </button>
  );
}

describe('Toast', () => {
  const toast = {
    title: 'test title',
    type: 'error',
    id: 'test-1',
  };
  const defaultProps = {
    toast,
  };
  const getWrapper = props =>
    mount(
      <ToastProvider>
        <TestInvoker {...defaultProps} {...props} />
      </ToastProvider>
    );

  beforeAll(() => {
    const toastRoot = global.document.createElement('div');

    toastRoot.setAttribute('id', 'toast-root');
    global.document.body.appendChild(toastRoot);
  });

  afterAll(() => {
    global.document.getElementById('toast').remove();
  });

  it('should render correct toast', async () => {
    const wrapper = getWrapper();

    wrapper.find('#create').simulate('click');
    await updateWrapper(wrapper);

    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });
});
