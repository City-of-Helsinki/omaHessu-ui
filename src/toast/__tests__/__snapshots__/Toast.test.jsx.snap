// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render correct toast 1`] = `
<ToastProvider>
  <Portal
    containerInfo={
      <div
        id="toast-root"
      >
        <div
          class="toastContainer"
        >
          <div
            class="notification"
          >
            <section
              aria-atomic="true"
              aria-label="Notification"
              class="inline notification default error"
            >
              <div
                class="content"
              >
                <div
                  aria-level="2"
                  class="label"
                  role="heading"
                >
                  <svg
                    aria-hidden="true"
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik icon"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <path
                        d="M0 0h24v24H0z"
                      />
                      <path
                        d="M11.128 2.51a1 1 0 011.7-.07l.044.07 9.562 17a1 1 0 01-.796 1.487l-.076.003H2.438a1 1 0 01-.906-1.423l.035-.067 9.562-17zM12 5.04L4.147 19h15.705L12 5.04zM13 16v2h-2v-2h2zm0-6.5v5h-2v-5h2z"
                        fill="currentColor"
                      />
                    </g>
                  </svg>
                  test title
                </div>
                <div
                  class="body"
                >
                  <div
                    class="messageWrapper"
                  >
                    notification.defaultErrorText
                  </div>
                </div>
              </div>
              <button
                aria-label="notification.closeButtonText"
                class="close error"
                title="notification.closeButtonText"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <rect
                      height="24"
                      width="24"
                    />
                    <polygon
                      fill="currentColor"
                      points="18 7.5 13.5 12 18 16.5 16.5 18 12 13.5 7.5 18 6 16.5 10.5 12 6 7.5 7.5 6 12 10.5 16.5 6"
                    />
                  </g>
                </svg>
              </button>
            </section>
          </div>
        </div>
      </div>
    }
  >
    <div
      className="toastContainer"
    >
      <Toast
        hidden={false}
        id="test-1"
        key="test-1"
        onClose={[Function]}
        title="test title"
        type="error"
      >
        <NotificationComponent
          labelText="test title"
          onClose={[Function]}
          show={true}
          type="error"
        >
          <div
            className="notification"
          >
            <ForwardRef
              closeButtonLabelText="notification.closeButtonText"
              dismissible={true}
              label="test title"
              onClose={[Function]}
              type="error"
            >
              <ConditionalVisuallyHidden
                visuallyHidden={false}
              >
                <section
                  aria-atomic="true"
                  aria-label="Notification"
                  className="inline notification default error"
                >
                  <div
                    className="content"
                  >
                    <div
                      aria-level={2}
                      className="label"
                      role="heading"
                    >
                      <Component
                        aria-hidden={true}
                        className="icon"
                      >
                        <svg
                          aria-hidden={true}
                          className="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik icon"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fillRule="evenodd"
                          >
                            <path
                              d="M0 0h24v24H0z"
                            />
                            <path
                              d="M11.128 2.51a1 1 0 011.7-.07l.044.07 9.562 17a1 1 0 01-.796 1.487l-.076.003H2.438a1 1 0 01-.906-1.423l.035-.067 9.562-17zM12 5.04L4.147 19h15.705L12 5.04zM13 16v2h-2v-2h2zm0-6.5v5h-2v-5h2z"
                              fill="currentColor"
                            />
                          </g>
                        </svg>
                      </Component>
                      <ConditionalVisuallyHidden
                        visuallyHidden={false}
                      >
                        test title
                      </ConditionalVisuallyHidden>
                    </div>
                    <div
                      className="body"
                    >
                      <div
                        className="messageWrapper"
                      >
                        notification.defaultErrorText
                      </div>
                    </div>
                  </div>
                  <button
                    aria-label="notification.closeButtonText"
                    className="close error"
                    onClick={[Function]}
                    title="notification.closeButtonText"
                    type="button"
                  >
                    <D
                      aria-hidden={true}
                    >
                      <svg
                        aria-hidden={true}
                        className="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fillRule="evenodd"
                        >
                          <rect
                            height="24"
                            width="24"
                          />
                          <polygon
                            fill="currentColor"
                            points="18 7.5 13.5 12 18 16.5 16.5 18 12 13.5 7.5 18 6 16.5 10.5 12 6 7.5 7.5 6 12 10.5 16.5 6"
                          />
                        </g>
                      </svg>
                    </D>
                  </button>
                </section>
              </ConditionalVisuallyHidden>
            </ForwardRef>
          </div>
        </NotificationComponent>
      </Toast>
    </div>
  </Portal>
  <TestInvoker
    toast={
      Object {
        "id": "test-1",
        "title": "test title",
        "type": "error",
      }
    }
  >
    <button
      id="create"
      onClick={[Function]}
    >
      test create
    </button>
  </TestInvoker>
</ToastProvider>
`;
